examples/src/picking.ts

```diff
@@ -0,0 +1,74 @@
+/*!
+ * Copyright 2020 Cognite AS
+ */
+
+import * as THREE from 'three';
+import CameraControls from 'camera-controls';
+import * as reveal from '@cognite/reveal';
+import * as reveal_threejs from '@cognite/reveal/threejs';
+
+CameraControls.install({ THREE });
+
+async function main() {
+  const modelUrl = new URL(location.href).searchParams.get('model') || '/primitives';
+
+  const pickedNodes: number[] = [];
+  const shading = reveal_threejs.createDefaultShading({
+    color(treeIndex: number) {
+      if (pickedNodes.indexOf(treeIndex) !== -1) {
+        return [255, 255, 0, 255];
+      }
+      return undefined;
+    }
+  });
+
+  const scene = new THREE.Scene();
+  const cadModel = await reveal.createLocalCadModel(modelUrl);
+  const cadNode = new reveal_threejs.CadNode(cadModel, { shading });
+
+  scene.add(cadNode);
+
+  const renderer = new THREE.WebGLRenderer();
+  renderer.setClearColor('#444');
+  renderer.setSize(window.innerWidth, window.innerHeight);
+  document.body.appendChild(renderer.domElement);
+
+  const { position, target, near, far } = cadNode.suggestCameraConfig();
+  const camera = new THREE.PerspectiveCamera(75, window.innerWidth / window.innerHeight, near, far);
+  const controls = new CameraControls(camera, renderer.domElement);
+  controls.setLookAt(position.x, position.y, position.z, target.x, target.y, target.z);
+  controls.update(0.0);
+  camera.updateMatrixWorld();
+  let pickingNeedsUpdate = false;
```

Comment:
Is there a way we can avoid this flag? It feels wrong since picking works on CadNode which already returns a flag from `update`.

Reply:
The flag is there because we need to update the scene after updating the colors. Think of it as "some user input changed the scene and we need to re-render". One way to integrate this more would be to add a listener in `CadNode` on some callback on `Shading` when `updateNodes` is called. What do you think?